modules/apps/29-fee/ibc_middleware.go

```diff
@@ -322,7 +322,31 @@ func (im IBCMiddleware) OnTimeoutPacket(
 
 // OnChanUpgradeInit implements the IBCModule interface
 func (im IBCMiddleware) OnChanUpgradeInit(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, sequence uint64, version, previousVersion string) (string, error) {
-	return im.app.OnChanUpgradeInit(ctx, portID, channelID, order, connectionHops, sequence, version, previousVersion)
+	var versionMetadata types.Metadata
+	if err := types.ModuleCdc.UnmarshalJSON([]byte(version), &versionMetadata); err != nil {
+		// Since it is valid for fee version to not be specified, the above upgrade version may be for a middleware
+		// or application further down in the stack. Thus, if it is not a fee version we pass the entire version string onto the underlying application.
+		return im.app.OnChanUpgradeInit(ctx, portID, channelID, order, connectionHops, sequence, version, previousVersion)
+	}
```

Comment:
We should ensure fee is disabled in this case, it might be nice to add some descriptive bool:

```go
err := types.ModuleCdc.UnmarshalJSON([]byte(version), &versionMetadata)
disableFees := err != nil

if disableFees {
    im.keeper.SetFeeDisabled(ctx, portID, channelID)
    return im.app.OnChanUpgradeInit(ctx, portID, channelID, order, connectionHops, sequence, version, previousVersion)
}

Reply:
changed the code structure a little by adding `types.MetadataFromVersion()` func, and now disabling fees on error